lib/auth/routes_authtype.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015-2018 _floragunn_ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+/*
+ * Portions Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+module.exports = function (pluginRoot, server, kbnServer, APP_ROOT, API_ROOT) {
+
+  /**
+   * Auth type API that returns current auth type configured on Kibana Server.
+   *
+   * GET /_plugin/kibana/_opendistro/api/authtype
+   * Response:
+   *  200 OK
+   *  {
+   *    authtype: saml
+   *  }
+   */
+   server.route({
+       method: 'GET',
+       path: `/_plugin/kibana/_opendistro/api/authtype`,
```

Comment:
why do we need `/_plugin/kibana` prefix?

Reply:
Wanted to use `/_plugin/kibana` since this is a whitelisted prefix.

Reply:
"_plugin/kibana" is an additional setting in kibana.yml. 

There is a different way to access that value. Most like "API_ROOT" should contain it.  
Example: https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/6eb41d3665c4153052f9071473043065d223fe5d/lib/auth/routes_authinfo.js#L36

Reply:
Thanks @dinusX @zengyan-amazon, renamed the API.